src/clj/game/cards/programs.clj

```diff
@@ -493,8 +493,9 @@
                                              (all-subs-broken-by-card? (:ice context) card)))
                               :msg (msg "add " (:title (:ice context))
                                         " to HQ after breaking all its subroutines")
-                              :effect (effect (move :corp (:ice context) :hand nil)
-                                              (continue :runner nil))}]}))
+                              :effect (req (set-current-ice state (:ice context))
```

Comment:
How is the context ice not already the current ice? That seems to be the real source of the bug.

Reply:
Fair. I changed `runs.clj` to update the current ice after subs get broken, and it appears to have not broken anything.